src/commands/test/lib/xcuitest-preparer.ts

```diff
@@ -0,0 +1,78 @@
+import * as fs from "fs";
+import * as iba from "../../../util/misc/ios-bundle-archiver";
+import * as path from "path";
+import * as pfs from "../../../util/misc/promisfied-fs";
+import * as pglob from "../../../util/misc/promisfied-glob";
+import { TestCloudError } from "./test-cloud-error";
+
+export class XCUITestPreparer {
+  private readonly artifactsDir: string;
+  private readonly buildDir: string;
+  private testIpaPath: string;
+
+  constructor(artifactsDir: string, buildDir: string, testIpaPath: string) {
+    if (!artifactsDir) {
+      throw new Error("Argument artifactsDir is required");
+    }
+
+    if (!(buildDir || testIpaPath)) {
+      throw new Error("Either buildDir or testIpaPath argument is required");
+    }
+
+    if (buildDir && testIpaPath) {
```

Comment:
Well, I'll just go on the record saying I think it's ambiguous. We have have (many) cases of people uploading the AUT in place of the runner, and the term `test-app/ipa/apk` seems inherently ambiguous when one app is the "tester" and the other is the "tested". 

For me, personally, I wouldn't know which `test-ipa-path` should refer to without some additional description. Maybe I'm alone in that opinion, but just thought I'd express it.